
# HG changeset patch
# User Mike Hommey <mh+mozilla@glandium.org>
# Date 1728937349 0
# Node ID d369655cf80acb2b00ba5c0dbf78792714b60a51
# Parent  c06005b986f807488a510aa788f3f2790540403a
Bug 1924022 - Use urlopen's context parameter instead of cafile. r=firefox-build-system-reviewers,jcristau,ahochheiden

The latter was removed in python 3.13.

Differential Revision: https://phabricator.services.mozilla.com/D225318

diff --git a/python/mozboot/mozboot/util.py b/python/mozboot/mozboot/util.py
--- a/python/mozboot/mozboot/util.py
+++ b/python/mozboot/mozboot/util.py
@@ -1,14 +1,15 @@
 # This Source Code Form is subject to the terms of the Mozilla Public
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
 import hashlib
 import os
+import ssl
 from pathlib import Path
 from urllib.request import urlopen
 
 import certifi
 from mach.site import PythonVirtualenv
 from mach.util import get_state_dir
 
 MINIMUM_RUST_VERSION = "1.76.0"
@@ -29,17 +30,18 @@ def get_mach_virtualenv_binary():
     return Path(PythonVirtualenv(str(root)).python_path)
 
 
 def http_download_and_save(url, dest: Path, hexhash, digest="sha256"):
     """Download the given url and save it to dest.  hexhash is a checksum
     that will be used to validate the downloaded file using the given
     digest algorithm.  The value of digest can be any value accepted by
     hashlib.new.  The default digest used is 'sha256'."""
-    f = urlopen(url, cafile=certifi.where())
+    ssl_context = ssl.create_default_context(cafile=certifi.where())
+    f = urlopen(url, context=ssl_context)
     h = hashlib.new(digest)
     with open(dest, "wb") as out:
         while True:
             data = f.read(4096)
             if data:
                 out.write(data)
                 h.update(data)
             else:

